.github/workflows/unit-tests.yaml

```
@@ -0,0 +1,33 @@
+name: CI/CD Action
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```

```suggestion
on:
  push:
  pull_request:
  workflow_dispatch:
```

Run CI on PR to all branches and manually on request as well.

---

This makes sense, thank you.